### PR TITLE
[DPP-739][Self-service error codes] Adapt QueueBackedTracker error codes

### PR DESCRIPTION
**Pull request by tudor-da**
This PR adapts the QueueBackedTracker status codes.

**NOTE**: Error codes migration guide adapted as well: https://htmlpreview.github.io/?https://github.com/tudor-da/docs-daml/blob/master/gen/index.html 

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala [hunk 1]
```diff
@@ -14,10 +14,72 @@ import com.daml.lf.language.{LanguageVersion, LookupError, Reference}
...
+        ) {
+      case class Reject(message: String, override val definiteAnswer: Boolean)(implicit
+          loggingContext: ContextualizedErrorLogger
+      ) extends LoggingTransactionErrorImpl(
+            cause = message
+          )
+    }
+  }
+
+  object CommandErrors extends ErrorGroup() {
```
Comment (tudor-da) [inline ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala:63]: Too generic.

File: ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala [hunk 2]
```diff
@@ -14,10 +14,72 @@ import com.daml.lf.language.{LanguageVersion, LookupError, Reference}
...
+import org.slf4j.event.Level
 
 import java.time.{Duration, Instant}
 
 object LedgerApiErrors extends LedgerApiErrorGroup {
+  object SubmissionErrors extends ErrorGroup() {
+    @Explanation(
+      """This error occurs when the participant rejects a command due to excessive load.
+        |Load can be caused as follows:
+        |1. when commands are submitted to the participant through its ledger api,
```
Comment (pbatko-da) [inline ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala:26]:
```suggestion
        |1. when commands are submitted to the participant through its Ledger Api, 
```

File: ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala [hunk 3]
```diff
@@ -14,10 +14,72 @@ import com.daml.lf.language.{LanguageVersion, LookupError, Reference}
...
 import com.daml.lf.value.Value
 import com.daml.lf.{VersionRange, language}
+import org.slf4j.event.Level
 
 import java.time.{Duration, Instant}
 
 object LedgerApiErrors extends LedgerApiErrorGroup {
+  object SubmissionErrors extends ErrorGroup() {
+    @Explanation(
+      """This error occurs when the participant rejects a command due to excessive load.
```
Comment (pbatko-da) [inline ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala:24]:
```suggestion
      """This error occurs when a participant rejects a command due to excessive load.
```

**Review by pbatko-da — APPROVED** [review]

File: ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala (hunk 2 above)
Comment (tudor-da) [reply ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala:26]: Done

File: ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala [hunk 4]
```diff
@@ -14,10 +14,50 @@ import com.daml.lf.language.{LanguageVersion, LookupError, Reference}
...
 import com.daml.lf.value.Value
 import com.daml.lf.{VersionRange, language}
+import org.slf4j.event.Level
 
 import java.time.{Duration, Instant}
 
 object LedgerApiErrors extends LedgerApiErrorGroup {
+  @Explanation(
+    """This error occurs when a participant rejects a command due to excessive load.
+        |Load can be caused as follows:
```
Comment (mziolekda) [inline ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala:24]:
```suggestion
        |Load can be caused by the following factors:
```

**Review by mziolekda — APPROVED** [review]

File: ledger/ledger-api-common/src/test/suite/scala/com/digitalasset/platform/server/api/validation/ErrorFactoriesSpec.scala [hunk 5]
```diff
@@ -531,6 +639,27 @@ class ErrorFactoriesSpec
     assertV2Error(error(errorFactoriesV2))(v2_code, v2_message, v2_details)
   }
 
+  private def assertVersionedStatus(
+      error: ErrorFactories => Status
+  )(
+      v1_code: Code,
```
Comment (miklos-da) [inline ledger/ledger-api-common/src/test/suite/scala/com/digitalasset/platform/server/api/validation/ErrorFactoriesSpec.scala:645]: Please follow [Scala naming conventions](https://docs.scala-lang.org/style/naming-conventions.html) for arguments.

**Pull request merged**